File: Draw generators JS/vfs2_rookie_draw_generator.js
function generateDraws() {
    const bb_figures = ['BB-A', 'BB-B', 'BB-C', 'BB-D', 'BB-E', 'BB-F', 'BB-G', 'BB-H', 'BB-J', 'BB-K', 'BB-L']; // Массив с фигурами BB
    const bb_blocks = ['BB-01', 'BB-02']; // Массив с блоками BB

    let allElements = [...bb_figures, ...bb_blocks]; // Объединяем массивы
    let draws = []; // Массив для хранения жеребьевок

    while (draws.length < 5) {
        let totalCost = Math.random() < 0.5 ? 3 : 4; // Случайная сумма стоимости (3 или 4)
        let draw = [];
        let currentCost = 0;
        let usedElements = []; // Для отслеживания уже использованных элементов

        while (currentCost < totalCost && draw.length < allElements.length) {
            // Выбираем случайный элемент
            let randomIndex = Math.floor(Math.random() * allElements.length);
            let element = allElements[randomIndex];

            // Определяем стоимость элемента
            let cost = bb_figures.includes(element) ? 1 : 2;

            // Проверяем, можем ли добавить элемент в жеребьевку
            if (!usedElements.includes(element) && currentCost + cost <= totalCost) {
                draw.push(element);
                usedElements.push(element);
                currentCost += cost;
            }
        }

        // Проверяем, достигли ли мы нужной суммы стоимости
        if ((draw.filter(el => bb_figures.includes(el)).length) <= 3) {
            draws.push(draw)
        }
    }

    return draws;
}


const results = generateDraws();
for (let i = 0; i < results.length; i++) {
    console.log(results[i].join('; '))
}

File: Draw generators Python/vfs2_rookie_draw_generator.py
import random


def generate_draws():
    bb_figures = ['BB-A', 'BB-B', 'BB-C', 'BB-D', 'BB-E', 'BB-F', 'BB-G', 'BB-H', 'BB-J', 'BB-K',
                  'BB-L']  # Массив с фигурами BB
    bb_blocks = ['BB-01', 'BB-02']  # Массив с блоками BB

    all_elements = bb_figures + bb_blocks  # Объединяем массивы
    draws = []  # Массив для хранения жеребьевок

    while len(draws) < 5:
        total_cost = random.choice([3, 4])  # Случайная сумма стоимости (3 или 4)
        draw = []
        current_cost = 0
        used_elements = set()  # Для отслеживания уже использованных элементов

        while current_cost < total_cost and len(draw) < len(all_elements):
            # Выбираем случайный элемент
            element = random.choice(all_elements)

            # Определяем стоимость элемента
            cost = 1 if element in bb_figures else 2

            # Проверяем, можем ли добавить элемент в жеребьевку
            if element not in used_elements and current_cost + cost <= total_cost:
                draw.append(element)
                used_elements.add(element)
                current_cost += cost

        # Проверяем, достигли ли мы нужной суммы стоимости
        if len([el for el in draw if el in bb_figures]) <= 3:
            draws.append(draw)

    return draws


results = generate_draws()
for draw in results:
    print('; '.join(draw))
